elastic/kibana
# [Discover] Cancel long running request after navigating out from Discover

## Summary

Fixes #122916

This PR add `AbortController` call to cancel long running request, after react component did unmount.  

### Test

1. Navigate to Discover
2. Use kibana_sample_data_logs
3. Switch query language to Lucene
4. Use a regular expression to search like /.*a.*b.*/
5. Switch to Dashboard while the search is running
In the network tab of your browser note that the bsearch request is no longer completed
